src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -28,60 +31,85 @@ internal sealed class DotNetCoreProjectCompatibilityDetector : IDotNetCoreProjec
         // to are "partial" so tooling should generally work, new features may not have tooling.
         private static Version s_partialSupportedVersion = new Version(2, 1);
 
-        private const string LearnMoreFwlink = "https://go.microsoft.com/fwlink/?linkid=867264";
+        private const string SupportedLearnMoreFwlink = "https://go.microsoft.com/fwlink/?linkid= 866848";
+        private const string UnsupportedLearnMoreFwlink = "https://go.microsoft.com/fwlink/?linkid=866796";
         private const string SuppressDotNewCoreWarningKey = @"ManagedProjectSystem\SuppressDotNewCoreWarning";
+        private const string VersionCompatibilityFwlink = "http://billhie1/test.json";   // TODO: test link  "https://go.microsoft.com/fwlink/?linkid=866798";
+        private const string VersionDataFilename = "DotNetVersionCompatibility.json";
+        private const int CacheFileValidHours = 24;
 
         [Guid("9B164E40-C3A2-4363-9BC5-EB4039DEF653")]
         private class SVsSettingsPersistenceManager { }
 
         enum CompatibilityLevel
         {
-            Supported = 0,
-            Partial = 1,
-            NotSupported =2
+            Recommended = 0,
+            Supported = 1,
+            NotSupported = 2
         }
         [ImportingConstructor]
-        public DotNetCoreProjectCompatibilityDetector([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, Lazy<IProjectServiceAccessor> projectAccessor, 
-                                                      Lazy<IDialogServices> dialogServices, Lazy<IProjectThreadingService> threadHandling)
+        public DotNetCoreProjectCompatibilityDetector([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, Lazy<IProjectServiceAccessor> projectAccessor,
+                                                      Lazy<IDialogServices> dialogServices, Lazy<IProjectThreadingService> threadHandling, Lazy<IVsShellUtilitiesHelper> vsShellUtilitiesHelper,
+                                                      Lazy<IFileSystem> fileSystem)
         {
             _serviceProvider = serviceProvider;
             ProjectServiceAccessor = projectAccessor;
             DialogServices = dialogServices;
             ThreadHandling = threadHandling;
+            ShellUtilitiesHelper = vsShellUtilitiesHelper;
+            FileSystem = fileSystem;
         }
 
+        private readonly IServiceProvider _serviceProvider;
+        private Lazy<IProjectServiceAccessor> ProjectServiceAccessor { get; set; }
+        private Lazy<IDialogServices> DialogServices { get; set; }
+        private Lazy<IProjectThreadingService> ThreadHandling { get; set; }
+        private Lazy<IFileSystem> FileSystem { get; set; }
+        private Lazy<IVsShellUtilitiesHelper> ShellUtilitiesHelper { get; set; }
```

Comment:
Why private properties instead of fields?  Do they all need "get" or can they be readonly?

Reply:
Just style, though I tend to use fields more now or properties with just a get. I'll fix this.